### ISSUE TITLE
Add in LocalScriptCreator

### ISSUE THREAD

**Issue by hlasimpk** [fsimkovic/pyjob#22]
Input 

- a function to generate scripts
- iterable options 
- the number of processors

Return 
- a ScriptCollector object.

Take the following example:

```
import os

from pyjob import pool
from pyjob.script import ScriptCollector, Script

nproc = 2
options = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
cwd = os.getcwd()


def get_script(option):
    cmd = ['echo {}'.format(option)]
    run_script = Script(directory=cwd)
    for c in cmd:
        run_script.append(' '.join(map(str, c)))
    return run_script


collector = ScriptCollector(None)
with pool.Pool(processes=nproc) as p:
    collector.add(p.map(get_script, options))
```

Replace this with something like:

```
import os

from pyjob import pool
from pyjob.script import LocalScriptCreator, Script

nproc = 2
options = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
cwd = os.getcwd()


def get_script(option):
    cmd = ['echo {}'.format(option)]
    run_script = Script(directory=cwd)
    for c in cmd:
        run_script.append(' '.join(map(str, c)))
    return run_script


collector = LocalScriptCreator(func=get_script, iterable=options, processes=nproc)
```

**Comment by hlasimpk** [comment]
I'm going to put in something to do this now

**Comment by fsimkovic** [comment]
I guess the rationale behind this is the automatic multi-"threaded" generation of `Script` instances in a `ScriptCollector` without the overhead of manually combining `Pool`, `Script`, and `ScriptCollector`.

**Comment by fsimkovic** [comment]
Implemented and incorporated in #23.
